src/intervals/rounding.jl

```diff
@@ -2,74 +2,191 @@
 # E.g.  +(a, b, RoundDown)
 # Some, like sin(a, RoundDown)  are already defined in CRlibm
 
+# Rounding types:
+# - :correct  # correct rounding
+# - :fast     # fast rounding by prevfloat and nextfloat  (slightly wider than needed)
+# - :none     # no rounding at all for speed
 
-import Base: +, -, *, /, sin, sqrt, inv, ^, zero, convert, parse
+# "Traits-based" design:
+# Define functions like
+# sin(::RoundingType{:correct}, x, ::RoundingMode)
+# sin(::RoundingType{:fast}, x, ::RoundingMode)
+# sin(::RoundingType{:none}, x, ::RoundingMode)
 
-# unary minus:
--{T<:AbstractFloat}(a::T, ::RoundingMode) = -a  # ignore rounding
+# Then define e.g.
+# sin(x, r::RoundingMode) = sin(RoundingType{:correct}, x, r)
 
-# zero:
-zero{T<:AbstractFloat}(a::Interval{T}, ::RoundingMode) = zero(T)
-zero{T<:AbstractFloat}(::Type{T}, ::RoundingMode) = zero(T)
 
-convert(::Type{BigFloat}, x, rounding_mode::RoundingMode) = setrounding(BigFloat, rounding_mode) do
-    convert(BigFloat, x)
-end
+doc"""Choose rounding mode based on environment variable"""
+
+immutable RoundingType{T} end
```

Comment:
I guess there is no need to impose that `T<:Symbol`... Or is it worth imposing it?